src/sentry/api/helpers/group_index.py

```diff
@@ -207,3 +223,98 @@ def validate(self, attrs):
             raise serializers.ValidationError(
                 'Other attributes cannot be updated when discarding')
         return attrs
+
+
+class GroupIndexMixin(object):
+    def handle_discard(self, request, group_list, projects, user):
```

Comment:
also 🤔these could probably be helper functions rather than mixin methods. i think there will be a couple things where the mixin will be necessary, but lmk what your thoughts are on what you think is preferable

Reply:
I personally lean towards helper functions/classes over mixins. The inheritance chains on our endpoints is often pretty deep and sometimes wide. A standalone function/class makes it simpler to locate where a method is defined and what its dependencies/required state are.

Reply:
k updated to avoid the mixin at least for now